eip_operator/src/main.rs

```diff
@@ -535,6 +781,69 @@ async fn cleanup_pod(
     Ok(Action::await_change())
 }
 
+#[instrument(skip(ec2_client, eip_api, node), err)]
+async fn cleanup_node(
+    ec2_client: &Ec2Client,
+    eip_api: &Api<Eip>,
+    node: Arc<Node>,
+) -> Result<Action, Error> {
+    let node_name = node.metadata.name.as_ref().ok_or(Error::MissingNodeName)?;
+    event!(Level::INFO, node_name = %node_name, "Cleaning up node.");
+    let node_labels = node
+        .metadata
+        .labels
+        .as_ref()
+        .ok_or(Error::MissingNodeLabels)?;
+    let all_eips = eip_api.list(&ListParams::default()).await?.items;
+    let eip = all_eips
+        .into_iter()
+        .filter(|eip| {
+            eip.status
+                .as_ref()
+                .map_or(false, |status| status.private_ip_address.is_some())
+        })
+        .find(|eip| match eip.spec.selector {
```

Comment:
This predicate function appears to be identical to the one introduced in `apply_node`. Can we refactor it into a reusable function?

Reply:
yeah, there is a lot of refactoring i'd like to do, but i'd prefer to do that in a followup pr to keep this one as straightforward as possible